src/amo/components/UserProfileEdit/index.js

```diff
@@ -610,9 +655,79 @@ export class UserProfileEditBase extends React.Component<Props, State> {
                     i18n.gettext("Update user's profile")
                 )}
               </Button>
+              <Button
+                buttonType="neutral"
+                className="UserProfileEdit-button UserProfileEdit-delete-button"
+                disabled={!user}
+                onClick={this.onDeleteProfile}
+                puffy
+                type="button"
+              >
+                {isEditingCurrentUser ?
+                    i18n.gettext('Delete my profile') :
+                    i18n.gettext(`Delete user's profile`)}
+              </Button>
             </div>
           </div>
         </form>
+
+        {this.state.showProfileDeletionModal && (
+          <OverlayCard
+            className="UserProfileEdit-deletion-modal"
+            header={isEditingCurrentUser ? i18n.gettext(
+              `Attention: you are about to delete your profile. Are you sure?`
+            ) : i18n.gettext(
+              `Attention: you are about to delete a profile. Are you sure?`
+            )}
+            onEscapeOverlay={this.onCancelProfileDeletion}
+            visibleOnLoad
+          >
+            <p
+              // eslint-disable-next-line react/no-danger
+              dangerouslySetInnerHTML={
+                sanitizeHTML(i18n.sprintf(
+                  i18n.gettext(`If you confirm this
+                    %(startStrong)sirreversible action%(endStrong)s, the
+                    following data will be removed: profile picture,
+                    profile details (including username, email, display
+                    name, location, home page, biography, occupation) and
+                    notification preferences. Other data such as ratings
+                    and reviews will be anonymized.`
+                  ), {
+                    startStrong: '<strong>',
+                    endStrong: '</strong>',
+                  }
+                ), ['strong'])
+              }
+            />
+            {isEditingCurrentUser && (
```

Comment:
Shouldn't we also warn admins that they cannot delete profiles of other users who currently have add-ons? Or do we just expect them to know that?

Reply:
I was thinking that admins would be aware but maybe we could have a message for them.

Reply:
You say "maybe" above, but you didn't change it. I'm fine either way - just wanted to raise it as something to consider.